keyboards/handwired/set_st/rp2040/extra.h

```diff
@@ -0,0 +1,117 @@
+/* Copyright 2024 set_st
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#ifdef OLED_ENABLE
+
+bool oled_task_user(void) {
+    // Set cursor position
+    oled_set_cursor(0, 0);
+    // Switch on current active layer
+    switch (get_highest_layer(layer_state)) {
+        case BASE :
+            oled_write_ln_P("1:Base", false);
+            break;
+        case FN :
+            oled_write_ln_P("2:Functions", false);
+            break;
+        case MEDIA : 
+            oled_write_ln_P("3:Media", false);
+            break;
+        case LIGHT :
+            oled_write_ln_P("4:Lights controll", false);
+            break;
+    }
+    
+
+    oled_set_cursor(0, 1);
+    oled_write(PSTR("Keys: "), false);
+
+    os_variant_t os = detected_host_os();
```

Comment:
If the os detection feature isn't enable, this will cause errors during compilation.  The ifdef check will prevent that. 
```suggestion
#ifdef OS_DETECTION_ENABLE
    os_variant_t os = detected_host_os();
```